### PR TITLE
[10.x.x] Fix for MSAA resolve performance on iOS/Metal

### PR DESCRIPTION
**Pull request by manuele-bonanno**
### Purpose of this PR
resolving MSAA to the backbuffer on iOS using the URP is quite slower (~1.1ms) than using the built-in pipeline. Fixed this, in the forward renderer, by adding additional criteria when deciding if an explicit resolve is needed

ES issue, containing test project: https://fogbugz.unity3d.com/f/cases/1219054/

---
### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [x] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

Yamato:
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/universal%252Fbugfix%252Fcase-1219054-msaa-resolve-ios

---
### Comments to reviewers

IMPORTANT: this issue highlights a problem with the explicit resolve pass selection criteria. The current fix is Metal specific, but it is very likely that other platforms will have the same problem. So it should be investigated on other platforms to follow up on this

Before the change:

beforePerf

URP

After the change:

afterPerf

afterFrame

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/ForwardRenderer.cs [hunk 1]
```diff
@@ -526,7 +532,7 @@ bool RequiresIntermediateColorTexture(ref CameraData cameraData)
             int msaaSamples = cameraTargetDescriptor.msaaSamples;
             bool isScaledRender = !Mathf.Approximately(cameraData.renderScale, 1.0f);
             bool isCompatibleBackbufferTextureDimension = cameraTargetDescriptor.dimension == TextureDimension.Tex2D;
-            bool requiresExplicitMsaaResolve = msaaSamples > 1 && !SystemInfo.supportsMultisampleAutoResolve;
+            bool requiresExplicitMsaaResolve = msaaSamples > 1 && PlatformRequiresExplicitAutoResolve();
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/ForwardRenderer.cs:535]: what about deferred renderer?

Comment (manuele-bonanno) [reply com.unity.render-pipelines.universal/Runtime/ForwardRenderer.cs:535]: I did a search for usages of SystemInfo.supportsMultisampleAutoResolve. Both the deferred and the 2D renderer use it to set the Depth texture BindMS flag:

depthDescriptor.bindMS = msaaSamples > 1 && !SystemInfo.supportsMultisampleAutoResolve && (SystemInfo.supportsMultisampledTextures != 0);

so it is not the same issue as the one I am fixing here, but it should probably be investigated as the autoResolve flag couldn't be the ideal check on some platforms?

**Comment by manuele-bonanno** [comment]
the failed HDRP tests are also broken in the nightly build. This changeset is only touching URP package code so not impacted by it

**Comment by phi-lira** [comment]
When this PR has finished review please add `ready-to-merge` tag and I'll pick it up/merge it. The failures in HDRP CI are not related to this PR.

**Pull request merged**